gdal/port/cpl_odbc.cpp

```diff
@@ -112,10 +112,13 @@ int CPLODBCDriverInstaller::InstallDriver( const char* pszDriver,
         }
 
         // Try to install ODBC driver in new location.
-        if( FALSE == SQLInstallDriverEx(pszDriver, nullptr, m_szPathOut,
+        if( FALSE == SQLInstallDriverEx(pszDriver, pszEnvHome, m_szPathOut,
                                         ODBC_FILENAME_MAX, nullptr, fRequest,
                                         &m_nUsageCount) )
         {
+            // if installing the driver fails, we need to roll back the changes to ODBCSYSINI environment
+            // variable or all subsequent use of ODBC calls will fail
+            unsetenv( "ODBCSYSINI" );
```

Comment:
I think MSVC won't swallow this.  Instead, [_putenv](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/putenv-wputenv?view=vs-2019):
> You can remove a variable from the environment by specifying an empty value_string, or in other words, by specifying only varname=

Reply:
Thanks. Thinking a bit more about this, I'm wondering if I should store any existing value for the variable before we initially set it, and then restore that value...

Reply:
That would make it more polite indeed

Reply:
Thanks, implemented